Cabal/Cabal.cabal

```diff
@@ -375,6 +375,7 @@ library
     Distribution.Simple.Program.Types
     Distribution.Simple.Register
     Distribution.Simple.Setup
+    Distribution.Simple.ShowBuildInfo
```

Comment:
Why the functionality needs to be in `Distribution.Simple`, can't it be implemented solely in `cabal-install`. There should be a rationale why it cannot be, if there are legitimate reasons.

Reply:
> 
> 
> Why the functionality needs to be in `Distribution.Simple`, can't it be implemented solely in `cabal-install`. There should be a rationale why it cannot be, if there are legitimate reasons.

There is, according to @hvr and @DanielG, and this pr is obsolete since the Cabal part has already been merged. 
For discussions on cabal-install part, see: #6241

Reply:
Good.

Sad we need support in `Cabal` though, as we have to worry about `build-type: Custom` stuff.